library/core/src/intrinsics/mod.rs

```diff
@@ -1434,7 +1434,8 @@ pub fn abort() -> ! {
 #[rustc_intrinsic_const_stable_indirect]
 #[rustc_nounwind]
 #[rustc_intrinsic]
-#[rustc_intrinsic_must_be_overridden]
+#[cfg_attr(bootstrap, rustc_intrinsic_must_be_overridden)]
+#[cfg_attr(not(bootstrap), rustc_intrinsic_lowers_to_mir)]
```

Comment:
I think I would prefer not requiring any annotations for intrinsics that are actually intrinsics.  Why should the library care whether the compiler implements these via lowering to MIR or in the backend? All intrinsics really are is a way to add more primitive operations to MIR without having to extend the AST. For some operations we keep the "function call" representation all the way to the backend, for some we have a more direct MIR representation -- but that's a compiler-internal implementation detail.

Reply:
> intrinsics are all compiler-internal implementation details, really

They are all internal to this repo, but they are not internal to `compiler/`. Their names and types are shared with `library/`. But the way they get lowered is private to the compiler, and exposing that in the library seems odd to me.

For stable MIR these are function calls, I don't understand that part of your comment.

Reply:
> I think it would be nice for backends to be able to tell what they need to handle, though, and not write implementations of things that aren't needed.

Do backend people read this file to figure that out? I would assume they run a test suite, which then tells them quite clearly what they need to implement. That's how I did it for Miri, anyway.